lib/jsonrpc.py

```diff
@@ -438,12 +438,16 @@ def decode_message(self, payload):
 
         Messages that cannot be decoded are logged and dropped.
         '''
-        try:
-            payload = payload.decode()
-        except UnicodeDecodeError as e:
-            msg = 'cannot decode message: {}'.format(e)
-            self.send_error(msg, JSONRPC.PARSE_ERROR)
-            return
+        encodings = ["utf-8", "iso-8859-1", 'cp-1251']
```

Comment:
JSON is defined as allowing only unicode encoding (RFC 4627). Adding additional server overhead to deal with faulty implementations seems wasteful.

Reply:
I've tested this on electrum-dash-old client (which works on python2) but electrumx can't parse json, so for me it was a solution of a problem